internal/dag/builder.go

```diff
@@ -99,22 +100,22 @@ func (b *Builder) lookupService(m k8s.FullName, port intstr.IntOrString) *Servic
 
 	s := lookup()
 	if s != nil {
-		return s
+		return s, nil
 	}
 	svc, ok := b.Source.services[m]
 	if !ok {
-		return nil
+		return nil, errors.New("service not found")
```

Comment:
This will result in some stutter in the error message, like "service foo error: service not found". That's not so bad, but maybe it's ok to just return "not found" here? Alternatively, we could move the quoted service name to this error?